pkg/api/v1/types.go

```diff
@@ -289,3 +292,63 @@ func NewVMReferenceFromName(name string) *VM {
 	vm.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
 	return vm
 }
+
+//TODO validate that this is correct
+func NewMigrationReferenceFromName(name string) *Migration {
+	migration := &Migration{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: kubeapi.NamespaceDefault,
+			SelfLink:  fmt.Sprintf("/apis/%s/namespaces/%s/%s", GroupVersion.String(), kubeapi.NamespaceDefault, name),
+		},
+	}
+	migration.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
+	return migration
+}
+
+type Migration struct {
+	metav1.TypeMeta `json:",inline"`
+	ObjectMeta      v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec            MigrationSpec `json:"spec,omitempty" valid:"required"`
+	Status          string        `json:"message,omitempty"`
+}
+
+// VMSpec is a description of a VM
+type MigrationSpec struct {
+	MigratingVMName         string            `json:"nodeName,omitempty"`
```

Comment:
I was thinking about a selector section inside the spec, but good enough for now. We will  see later on if it makes sense to allow selecting the vm also by labels, or even allow to select multiple VMs.